pkg/filters/proxy/basepool.go

```diff
@@ -125,7 +132,11 @@ func (bsp *BaseServerPool) createLoadBalancer(spec *LoadBalanceSpec, servers []*
 	}
 
 	lb := NewLoadBalancer(spec, servers)
+	old := bsp.LoadBalancer()
 	bsp.loadBalancer.Store(lb)
+	if old != nil {
+		old.Close()
+	}
```

Comment:
please use [SwapPointer](https://pkg.go.dev/sync/atomic#SwapPointer) to refactor L135-139.